pkg/api/types.go

```diff
@@ -2233,6 +2233,10 @@ type Secret struct {
 }
 
 const MaxSecretSize = 1 * 1024 * 1024
+const MaxSecretKeys = 1024
+
+// reserve 1k from the MaxSecretSize for serialization
+const MaxUnserializedSecretSize = MaxSecretSize - 1024
```

Comment:
I really don't see the point of this.  The bias is a totally random guess, and is only "somewhat more likely" to be safe than not.  We do not have a systemic check for max serialized object size, so IMO, we should not be doing ad hoc checks like this.  It adds complexity for very little win.  If the bias were 25% or something (and applied to other objects with payloads), maybe, but 1k?